src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,46 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            try
+            {
+                var serviceProvider = ExtensionServiceProvider.Create(new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) });
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    var cancellationToken = new CancellationTokenSource(ExtensionLoadingTimeout).Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken);
+                    if (ext == null)
+                    {
+                        await requestContext.SendError("Failed to create ICompletionExtension");
```

Comment:
Do we need to localize these?

Reply:
I'm not sure, what's the normal practice on error messages?

Reply:
If it's possible to show up as an error / warning popup to the user, needs localization. if it's mostly diagnostic but _might_ show in the Output area I think it can skip localization

Reply:
This should be diagnostic message to output and telemetry, shouldn't show up as a error/warning popup to the user. So we don't need localization here.